compiler/src/dotty/tools/dotc/ast/Desugar.scala

```diff
@@ -886,7 +888,7 @@ object desugar {
           mdef.tparams.head.srcPos)
       defDef(
         cpy.DefDef(mdef)(
-          name = mdef.name.toExtensionName,
+          name = normalizeName(mdef, ext).toExtensionName,
```

Comment:
@bishabosha Moved the method name check here (that disallows extension method definitions starting with `extension_`), instead of its original place in `normalizeName`. The reason is that originally the warning was a bit misleading:
* before
```
8 |extension (x: Any) def extension_foo: String = "foo" // error: illegal name: extension_foo may not start with `extension_`
  |                       ^^^^^^^^^^^^^
  |                       illegal name: extension_extension_foo may not start with `extension_`
```
* after:
```
8 |extension (x: Any) def extension_foo: String = "foo" // error: illegal name: extension_foo may not start with `extension_`
  |                       ^^^^^^^^^^^^^
  |                       illegal name: extension_foo may not start with `extension_`
```